x-pack/platform/plugins/shared/onechat/server/services/tools/esql/storage.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger, ElasticsearchClient } from '@kbn/core/server';
+import { ToolDescriptorMeta } from '@kbn/onechat-common';
+import { IndexStorageSettings, StorageIndexAdapter, types } from '@kbn/storage-adapter';
+
+export const esqlToolIndexName = '.kibana_onechat_esql_tools';
+
+const storageSettings = {
+  name: esqlToolIndexName,
+  schema: {
+    properties: {
+      id: types.keyword({}),
+      name: types.text({}),
```

Comment:
NIT: My gut feeling would be to use `keyword` for `name` here, not `text`, as search needs for such fields are usually keyword-based. Only an opinion though.